src/components/Toolbar.ts

```diff
@@ -74,6 +81,52 @@ const toolbarView = (vnode: Vnode<IToolbar>) => {
           },
           "Import HAR Session"
         ),
+        m(
+          "a.button.is-outlined.is-small.control",
+          {
+            class: ngrokStreamLock === -1 ? "ngrok-play" : "ngrok-stop"
+          },
+          m("hi", {
+            onclick: () => {
+            if (ngrokStreamLock != -1) {
+              console.log('stopping ngrok stream');
+              ngrokStreamLock = -1;
+              return;
+            }
+
+            // this permissions things works but it needs a better way
+            // to test and reset it!
+
+            var API_HOST = 'http://localhost:4040/'; // TODO: make this customisable in settings
```

Comment:
For stuff like this it's usually just a call to `chrome.storage`, if you don't want to put in in the UI here you can just hook it up in `options.ts`.